confgenerator/confgenerator_test.go

```diff
@@ -48,48 +45,88 @@ var (
 	invalidInputPath   = invalidTestdataDir + "/%s/%s/input.yaml"
 )
 
-var platform string
-var hostInfo *host.InfoStat
+type platformConfig struct {
+	defaultLogsDir  string
+	defaultStateDir string
+	defaultConfig   string
+	*host.InfoStat
+}
+
+var platforms = []platformConfig{
+	platformConfig{
+		defaultLogsDir:  "/var/log/google-cloud-ops-agent/subagents",
+		defaultStateDir: "/var/lib/google-cloud-ops-agent/fluent-bit",
+		defaultConfig:   "default-config.yaml",
+		InfoStat: &host.InfoStat{
+			OS:              "linux",
+			Platform:        "linux_platform",
+			PlatformVersion: "linux_platform_version",
+		},
+	},
+	platformConfig{
+		defaultLogsDir:  `C:\ProgramData\Google\Cloud Operations\Ops Agent\log`,
+		defaultStateDir: `C:\ProgramData\Google\Cloud Operations\Ops Agent\run`,
+		defaultConfig:   "windows-default-config.yaml",
+		InfoStat: &host.InfoStat{
+			OS:              "windows",
+			Platform:        "win_platform",
+			PlatformVersion: "win_platform_version",
+		},
+	},
+}
 
 func init() {
-	hostInfo, _ = host.Info()
-	if hostInfo.OS == "windows" {
-		platform = "windows"
-		//In order to make test data static, we put static value for platform-wise fields.
-		hostInfo.Platform = "win_platform"
-		hostInfo.PlatformVersion = "win_platform_version"
-	} else {
-		platform = "linux"
-		//In order to make test data static, we put static value for platform-wise fields.
-		hostInfo.Platform = "linux_platform"
-		hostInfo.PlatformVersion = "linux_platform_version"
+	// filepathJoin is overriden for tests in order to
+	// impersonate the behavior of an alternate OS.
+	filepathJoin = func(goos string, elem ...string) string {
+		separator := "/"
+		if goos == "windows" {
+			separator = `\`
+		}
+		return strings.Join(elem, separator)
+	}
+}
+
+func TestDefaultFilepathJoin(t *testing.T) {
+	t.Parallel()
+
+	// Test that the default filepathJoin function does not
+	// generate paths that are dependent on the specified OS.
+	abc := filepath.Join("a", "b", "c")
+	linuxAbc := defaultFilepathJoin("linux", "a", "b", "c")
+	windowsAbc := defaultFilepathJoin("windows", "a", "b", "c")
+	if abc != linuxAbc || abc != windowsAbc {
```

Comment:
Nice!

[Optional] The error could indicate which of the two doesn't match the behavior.
Or, if you switch to `Error`, you could split it into two checks, e.g.:
```
	if abc != linuxAbc {
		t.Errorf(`defaultFilepathJoin("linux")` does not match filepath.Join: %q`, linuxAbc)
	}
	if abc != windowsAbc {
		t.Errorf(`defaultFilepathJoin("windows")` does not match filepath.Join: %q`, windowsAbc)
	}
```